hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/services/TestAbfsRenameRetryRecovery.java

```diff
@@ -123,6 +138,112 @@ public void testRenameFailuresDueToIncompleteMetadata() throws Exception {
 
   }
 
+  AbfsClient getMockAbfsClient() throws IOException {
+    AzureBlobFileSystem fs = getFileSystem();
+
+    // specifying AbfsHttpOperation mock behavior
+
+    // mock object representing the 404 path not found result
+    AbfsHttpOperation mockHttp404Op = Mockito.mock(AbfsHttpOperation.class);
```

Comment:
It would be awesome if we can use server integration in picture. Have added a comment on @sreeb-msft 's pr https://github.com/apache/hadoop/pull/5488#discussion_r1141679607. Suggested changes in https://github.com/saxenapranav/hadoop/commit/5247e12bf2b96f8397d4c6df0865117a18dd13fd